### PR TITLE
Add DNSHostname to NetAddress mssgs

### LINKED ISSUE #1313: Support DNS hostnames in gossip
See https://github.com/lightning/bolts/pull/911

### PR DESCRIPTION
**Pull request by mauricepoirrier**
Support Bolt 7 DNS Hostnames Check https://github.com/lightning/bolts/pull/911

I'm missing the test in this one if someone could guide me would be nice

Fixes #1313

### CODE REVIEW

File: lightning/src/ln/msgs.rs [hunk 1]
```diff
@@ -434,6 +434,14 @@ pub enum NetAddress {
 		/// The port on which the node is listening
 		port: u16,
 	},
+	DNSHostname {
+		/// The hostname length
+		hostname_len: u8,
```
Comment (TheBlueMatt) [inline lightning/src/ln/msgs.rs:439]: You don't need an explicit length if its implicit in the Vec below :).

File: lightning/src/ln/msgs.rs [hunk 2]
```diff
@@ -434,6 +434,14 @@ pub enum NetAddress {
...
+		/// The dns hostname on which the node is listening, length hostname len 
+		hostname: Vec<u8>,
```
Comment (TheBlueMatt) [inline lightning/src/ln/msgs.rs:441]: Probably just make this a String, no?

File: lightning/src/ln/msgs.rs [hunk 3]
```diff
@@ -454,11 +463,12 @@ impl NetAddress {
 			&NetAddress::IPv6 { .. } => { 18 },
 			&NetAddress::OnionV2(_) => { 12 },
 			&NetAddress::OnionV3 { .. } => { 37 },
+			&NetAddress::DNSHostname { .. } => { 258 }
```
Comment (TheBlueMatt) [inline lightning/src/ln/msgs.rs:466]: This should return the real serialized length, using the length of the hostname itself.

**Review by TheBlueMatt — COMMENTED** [review]
Thanks for taking a crack at this!

**Comment by mauricepoirrier** [comment]
@TheBlueMatt sorry the delay, I changed the comments.
Kinda learning rust with these. I hope looks better :)

File: lightning/src/ln/msgs.rs [hunk 4]
```diff
@@ -434,6 +435,12 @@ pub enum NetAddress {
 		/// The port on which the node is listening
 		port: u16,
 	},
+	DNSHostname {
+		/// The dns hostname on which the node is listening, length hostname len
```
Comment (TheBlueMatt) [inline lightning/src/ln/msgs.rs:439]: s/, length hostname len //?

File: lightning/src/ln/msgs.rs [hunk 5]
```diff
@@ -485,6 +496,11 @@ impl Writeable for NetAddress {
 				version.write(writer)?;
 				port.write(writer)?;
 			}
+			&NetAddress::DNSHostname { ref hostname, ref port } => {
+				5u8.write(writer)?;
+				hostname.write(writer)?;
```
Comment (TheBlueMatt) [inline lightning/src/ln/msgs.rs:501]: The default `String` write writes a two-byte length, whereas we need one here. You'll need to manually write the string. That said, you may actually want to use a different type, given we're required to write out hostnames in ASCII, and the tight size length, we'll want to enforce that on users.

File: lightning/src/ln/msgs.rs [hunk 6]
```diff
@@ -515,6 +531,13 @@ impl Readable for Result<NetAddress, u8> {
 					port: Readable::read(reader)?,
 				}))
 			},
+			5 => {
+				Readable::read(reader)?;
+				Ok(Ok(NetAddress::DNSHostname {
+					hostname: Readable::read(reader)?,
```
Comment (TheBlueMatt) [inline lightning/src/ln/msgs.rs:537]: We should enforce the ASCII-ness of the string here.

**Comment by TheBlueMatt** [comment]
Do you intend to revisit this @mauricepoirrier?

**Comment by mauricepoirrier** [comment]
> Do you intend to revisit this @mauricepoirrier?

Yes @TheBlueMatt. I was thinking on this weekend.

**Pull request closed without merging**